poollet/volumepoollet/vcm/generic.go

```diff
@@ -53,6 +57,41 @@ type Generic struct {
 	relistPeriod time.Duration
 }
 
+func (g *Generic) AddNotifier(classNotifier orievent.Notifier) (orievent.NotifierRegistration, error) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+
+	h := &notifier{Notifier: classNotifier}
+
+	g.notifier.Insert(h)
+	return &notifierRegistration{
+		vcm:      g,
+		notifier: h,
+	}, nil
+}
+
+func (g *Generic) notify(classes []*ori.VolumeClass) {
+	oldClasses := maps.Keys(g.volumeClassByName)
+	if len(classes) != len(oldClasses) {
+		return
+	}
+
+	sameNames := true
+	for _, class := range classes {
+		if _, found := g.volumeClassByName[class.Name]; !found {
+			sameNames = false
+			break
+		}
+	}
+	if sameNames {
+		return
+	}
+
+	for _, n := range g.notifier.UnsortedList() {
+		n.Notify()
```

Comment:
There might be problems here with locks / synchronisation since we're calling `Notify` while holding the lock. Notification should be done similar as e.g. in `orievent.generator` where the swapping of values is atomic and events are computed afterwards.